api/src/main/java/org/codice/ddf/admin/api/handler/MessageBuilder.java

```diff
@@ -0,0 +1,36 @@
+package org.codice.ddf.admin.api.handler;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+public class MessageBuilder {
+
+    private Map<String, String> successTypes;
+    private Map<String, String> failureTypes;
+    private Map<String, String> warningTypes;
+    private Map<String, String> allTypes;
+
+    public MessageBuilder(Map<String, String> successTypes, Map<String, String> failureTypes, Map<String, String> warningTypes){
+        this.successTypes = ImmutableMap.<String, String>builder().putAll(successTypes).build();
+        this.failureTypes = ImmutableMap.<String, String>builder().putAll(failureTypes).build();
+        this.warningTypes = ImmutableMap.<String, String>builder().putAll(warningTypes).build();
+        this.allTypes = ImmutableMap.<String, String>builder().putAll(successTypes).putAll(failureTypes).putAll(warningTypes).build();
+    }
+
+    public Map<String, String> getDescriptions(String... subtypeKeys) {
+        Map<String, String> descriptions = new HashMap<>();
```

Comment:
I had an issue with this simplification blowing up. I'm going to leave it as is for now. I'm not sure if it's a issue with Function.identity or what, but it broke the entire deployment. I didn't figure out what was going on after about 45 minutes and decided it wasn't worth the time at the moment.